centos/base/Dockerfile

```diff
@@ -0,0 +1,216 @@
+    && chown -R stanley:stanley /home/stanley/.ssh \
+    && echo "stanley    ALL=(ALL)       NOPASSWD: SETENV: ALL" >> /etc/sudoers.d/st2 \
+    && chmod 0440 /etc/sudoers.d/st2 \
+    && sed -i -r "s/^Defaults\s+\+?requiretty/# Defaults +requiretty/g" /etc/sudoers
+
+RUN dnf install dnf-plugins-core epel-release -y
+# Generates nginx ssl key
+RUN rpm --import http://nginx.org/keys/nginx_signing.key \
+    && yum-config-manager --add-repo http://nginx.org/packages/centos/7/x86_64 \
+    #&& dnf update -y \
+    && dnf --disablerepo='epel' install -y nginx \
+    && mkdir -p /etc/ssl/st2 \
+    && openssl req -x509 -newkey rsa:2048 -keyout /etc/ssl/st2/st2.key -out /etc/ssl/st2/st2.crt \
+      -days 365 -nodes -subj "/C=US/ST=California/L=Palo Alto/O=StackStorm/OU=Information \
+      Technology/CN=$(hostname)"
+
+# Copys all in one nginx config
+COPY conf/st2-base.cnf /etc/nginx/conf.d/st2-base.cnf
+COPY conf/nginx.conf /etc/nginx/nginx.conf
+
+# Setup 1ppc nginx
+COPY conf/st2-https.template /etc/nginx/conf.d/st2-https.template
+COPY conf/st2.conf-https.patch /tmp
+RUN patch /etc/nginx/conf.d/st2-https.template < /tmp/st2.conf-https.patch 
+
+# Systemd requires a SIGRTMIN+3 signal to terminate properly.
+STOPSIGNAL SIGRTMIN+3
+
+# Remove unnecessary systemd targets
+RUN (cd /lib/systemd/system/sysinit.target.wants/; for i in *; do [ $i == systemd-tmpfiles-setup.service ] || rm -f $i; done); \
+rm -f /lib/systemd/system/multi-user.target.wants/*;\
+rm -f /etc/systemd/system/*.wants/*;\
+rm -f /lib/systemd/system/local-fs.target.wants/*; \
+rm -f /lib/systemd/system/sockets.target.wants/*udev*; \
+rm -f /lib/systemd/system/sockets.target.wants/*initctl*; \
+rm -f /lib/systemd/system/basic.target.wants/*;\
+rm -f /lib/systemd/system/anaconda.target.wants/*;
+
+# named (dns server) service
+RUN systemctl enable named.service
+
+# Adds the run directory for systemd. This directory will need to run as a
+# tmpfs directory for systemd to properly work.
+VOLUME /run
+
+# Systemd enabling all st2 components to run at container start time.
+# Results tracker removed https://github.com/StackStorm/st2/issues/5070
+RUN systemctl enable \
+        st2actionrunner \
+        st2api \
+        st2auth \
+        st2garbagecollector \
+        st2notifier \
+        st2rulesengine \
+        st2sensorcontainer \
+        st2scheduler \
+        st2workflowengine \
+        st2timersengine \
+        st2workflowengine \
+        nginx
+
+# Systemd runs as user st2. This changes the log files to allow st2 to run
+RUN chown -R st2 /var/log/st2
+
+# Opens ssh and https ports.
+EXPOSE 22 443
+
+# Used by all stackstorm services
+VOLUME ["/etc/st2"]
+WORKDIR /opt/stackstorm
+
+COPY bin/entrypoint.sh /st2-docker/bin/entrypoint.sh
+
+ENTRYPOINT ["/st2-docker/bin/entrypoint.sh"]
+
+# For all-in-one docker-compose development
+RUN mkdir /opt/stackstorm/packs.dev
+
+# Update oslo/crudini
+RUN source /opt/stackstorm/st2/bin/activate && pip install --upgrade oslo.config
```

Comment:
Any more context behind updating the `oslo.config`?

Reply:
I think we needed this for crudini; I'll check again...